### PR TITLE
[Base] ADD: Check on already used field(s) in inherited views

### PR DESCRIPTION
**Pull request by Yenthe666**
Description of the issue/feature this PR addresses: You can add the same field to multiple views without ever getting any warning. This can have strange behaviour in combination with `api.onchange` though (see https://github.com/odoo/odoo/issues/15111).

Current behavior before PR: There is a warning in the logfile that shows you the field is already used and shown in another view.

Desired behavior after PR is merged: If a field is already in another inherited view there is a warning in the logger in the following format:
```
2017-02-06 10:51:07,227 7526 WARNING yourDbName odoo.addons.base.ir.ir_ui_view: The fields ['barcode'] are already present in the view definition (view name: product.template.product.form, view id: 320, model: product.template, tag in the view: notebook).
```

Usecase:
By default Odoo does not detect if a field is already defined in another view or not. This can however lead to strange issues when you want to add this same field to another view again and when you add an `@api.onchange` on this field. (For more information see https://github.com/odoo/odoo/issues/15111). 
An example usecase:
Imagine having a view named `product.template.product.form`. On this view you show the field `barcode` in some tab. After a few months you create a new view named `product.template.product.form.inherit` that inherits the already existing view `product.template.product.form`. For some reason you however forget that the original view already shows the `barcode` field and you add it in the new view.
In this case you will have the same field (`barcode`) twice in the view. Once from the view `product.template.product.form` and once from the view `product.template.product.form`. When you would now do an @api.onchange you would get very strange behaviour, not to mention that you basically didn't even need this field on the view, since it was already there.
This PR shows a warning in the logfile when this issue happens.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by mart-e** [comment]
This should target master (could you rebase?) and to be merged, will require fixing all the detected errors :wink:

File: odoo/addons/base/ir/ir_ui_view.py [hunk 1]
```diff
@@ -769,7 +769,13 @@ def postprocess(self, model, node, view_id, in_tree_view, model_fields):
 
         for f in node:
             if children or (node.tag == 'field' and f.tag in ('filter', 'separator')):
-                fields.update(self.postprocess(model, f, view_id, in_tree_view, model_fields))
+                postprocess_fields = self.postprocess(model, f, view_id, in_tree_view, model_fields)
+                union = list(set(fields) & set(postprocess_fields))
+                if union:
+                    # This will give the name of the view (record) in the database. It is better than just an ID. 
+                    view_name = self.env['ir.ui.view'].search([('id', '=', view_id)]).name
```
Comment (mart-e) [inline odoo/addons/base/ir/ir_ui_view.py]: use `browse` + `name_get` instead

**Comment by mart-e** [comment]
Actually I do not think we are going to merge this.
There is several issues with this code implementation that makes false positive:
```xml
<calendar string="Scheduled Actions" date_start="nextcall" color="user_id">
    <field name="user_id"/>
</calendar>
```
```xml
<field name="name" />
<field name="line_ids">
     <field name="name" />
</field>
```
```xml
<kanban>
  <field name="name" />
  <templates>
     <field name="name" />
  </templates>
</kanban>
```
are all valid views but raise a warning.

So a different detection needs to be done.

**Comment by Yenthe666** [comment]
Hi @mart-e,

Thanks for the further information.
I do agree with you that it are false positives but perhaps we can add checks on that in here? What other option do you suggest? 😄

**Comment by mart-e** [comment]
> What other option do you suggest?

Find a different check that is not making false positive for these. (and to answer your next question, no I do not know which test is that ;-) )

You can also use this code to detect duplicated fields (cf c2c91a1fdf4f889723d22c9059e1f5c40e505d17 and 496c77cb5a1ce24f1ab728925771f5b2140d90bc) but only for your testing, not to add in standard

**Pull request closed without merging**